hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/NativeSparkDataSet.java

```diff
@@ -1080,4 +1105,220 @@ public TableSamplerBuilder sample(OperationContext operationContext) throws Stan
+                        // Splice already bumps the scale up by 4 for AVG,
+                        // Spark does the same as well in class Average.
+                        // The redundant scale adjustment causes overflows,
+                        // which manifest as nulls in Spark.  If we attempt to
+                        // remove the redundant scale adjustment by applying a CAST,
+                        // we get truncated digits (e.g. 4 trailing zeroes in the
+                        // rightmost positions to the right of the decimal place).
+                        // To make results from the control path match results from
+                        // the Spark path we evaluate avg as sum/count.
+
+                        if (aggregates[i].isDistinct()) {
+                            Column dividend = sumDistinct(sourceColumn);
+                            Column divisor  = countDistinct(sourceColumn);
+                            column = dividend.divide(divisor).cast(targetDataType);
+                        }
+                        else {
+                            Column dividend = sum(sourceColumn);
+                            Column divisor  = org.apache.spark.sql.functions.count(sourceColumn);
+                            column = dividend.divide(divisor).cast(targetDataType);
+                        }
+                        break;
+                    case "COUNT":
+                        column = aggregates[i].isDistinct() ?
+                                  countDistinct(sourceColumn) :
+                                  org.apache.spark.sql.functions.count(sourceColumn);
+                        column = column.cast(targetDataType);
+                        inputColumns.add(sourceColName);
+                        break;
+                    case "COUNT(*)":
+                        // There is no COUNT(DISTINCT *) in the grammar.
+                        if (aggregates[i].isDistinct())
+                            return null;
+
+                        column = org.apache.spark.sql.functions.count(new Column("*"));
+                        column = column.cast(targetDataType);
+                        break;
+                    case "SpliceStddevPop":
+                        column = stddev_pop(sourceColName);
+                        column = column.cast(targetDataType);
+                        // Distinct not currently supported.  See SPLICE-1820.
+                        if (aggregates[i].isDistinct())
+                            return null;
+                        inputColumns.add(sourceColName);
+                        break;
+                    case "SpliceStddevSamp":
+                        // Disabling this path for now.
+                        // stddev_samp may return NaN unexpectedly,
+                        // which on Splice causes the query to error out.
+                        // See SPARK-13860.
+                        if (stddev_samp_Disabled)
```

Comment:
Could we compute stddev_samp from stdev_pop here? I think samp = sqrt(((pop^2)*count)/(count-1)) 
That's assuming stddev_pop handles NULLs gracefully.
I'm not sure how much precision we could lose, but since tpcds uses the function it would be good to have some native support.

Reply:
@dgomezferro It looks to be a divide by zero problem, so I added a 'when' function to return null when the divisor is zero:
`column = when(countColumn.gt(lit(1)), column);`